src/pool.rs

```diff
@@ -117,7 +122,7 @@ impl ConnectionPool {
     /// Attempts to acquire a connected socket. If none are available and
     /// the pool has not reached its maximum size, a new socket will connect.
     /// Otherwise, the function will block until a socket is returned to the pool.
-    pub fn acquire_stream(&self) -> Result<PooledStream> {
+    pub fn acquire_stream(&self, client: Client) -> Result<PooledStream> {
```

Comment:
In order to run `isMaster` with the client metadata after connecting, we need to have a reference to the Client. This unfortunately requires us to pass a reference to it any time `acquire_stream` is called, which percolates through a decent amount of the topology code. Luckily, none of the structs themselves needed to be updated to support this, and the only user-facing API (i.e. `Client#acquire_stream`) can just clone `self` to be passed in, so I think the user-facing changes here are minimal.